### PR TITLE
Feat/7966 search BTM => Improve search: add search entities and highlight the results

### PR DESCRIPTION
**Pull request by morhag90**
Close: #7966 

## PR Details

Improve search: add search entities and highlight the results.
## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/Project/ArchitectureConsole/types.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export type NodePayload<T> = {
   height?: number;
   originalParentNode?: string;
   isCurrentDropTarget?: boolean;
+  highlight: boolean;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/types.ts:16]: make this param options
```suggestion
  highlight?: boolean;
```

File: packages/amplication-client/src/Project/ArchitectureConsole/helpers.ts [hunk 2]
```diff
@@ -142,7 +143,11 @@ function entitiesToNodes(
 
   const nodes: EntityNode[] = resources.flatMap((resource) =>
     resource.entities.map((entity) => ({
-      data: { payload: entity, originalParentNode: entity.resourceId },
+      data: {
+        highlight: false,
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/helpers.ts:147]: we dont have to add this after we make the property optional

File: packages/amplication-client/src/Project/ArchitectureConsole/helpers.ts [hunk 3]
```diff
@@ -127,6 +127,7 @@ function entitiesToNodes(
       groupOrder: index,
       groupColor: resourceColorMapping[index],
       isEditable: false,
+      highlight: false,
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/helpers.ts:130]: we dont have to add this after we make the property optional

File: packages/amplication-client/src/Project/ArchitectureConsole/hooks/useModelOrganizer.tsx [hunk 4]
```diff
@@ -449,37 +449,23 @@ const useModelOrganization = ({ projectId, onMessage }: Props) => {
...
-          );
-
-          childrenNodes.forEach((x) => (x.hidden = true));
+      nodes.forEach((x) => (x.data.highlight = false));
 
-          const nodeEdges = edges.filter((e) => {
-            return childrenNodes.find((n) => e.source === n.id);
-          });
+      if (searchPhrase !== "") {
+        const searchModelGroupNodes = nodes.filter((node) =>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/hooks/useModelOrganizer.tsx:455]: these are not just modelGroupNodes
```suggestion
        const searchResults = nodes.filter((node) =>
```

File: packages/amplication-client/src/Project/ArchitectureConsole/nodes/modelGroupNode.tsx [hunk 5]
```diff
@@ -2,18 +2,17 @@ import { memo, type FC, useCallback } from "react";
 import { useStore, type NodeProps } from "reactflow";
 import "./modelGroupNode.scss";
 import {
-  EnumButtonStyle,
   EnumContentAlign,
   EnumFlexDirection,
   EnumItemsAlign,
+  EnumTextColor,
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/nodes/modelGroupNode.tsx:8]: can be removed

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Review by mulygottlieb — APPROVED** [review]
Approved, comments by Yuval were addressed.

**Pull request merged**